Spigot-Server-Patches/0565-Add-additional-open-container-api-to-HumanEntity.patch

```diff
@@ -0,0 +1,127 @@
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockCartographyTable) Blocks.CARTOGRAPHY_TABLE).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openGrindstone(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
++            if (block.getType() != Material.GRINDSTONE) {
++                return null;
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockGrindstone) Blocks.GRINDSTONE).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openLoom(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
++            if (block.getType() != Material.LOOM) {
++                return null;
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockLoom) Blocks.LOOM).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openSmithingTable(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
++            if (block.getType() != Material.SMITHING_TABLE) {
++                return null;
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockSmithingTable) Blocks.SMITHING_TABLE).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openStonecutter(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
```

Comment:
You check if`location` is `null` below, but not here - is this intentional? If so, why?